### DISCUSSION TITLE
Spelling error in cmake-proxies/cmake-modules/AudacityDependencies.cmake

### DISCUSSION THREAD

**Discussion by evertvorster** [audacity/audacity#3477] [Development]
### Bug description

wxWidget is spelled wxWdidget about 25% down the file. 
I'm pretty sure it's a spelling error.

it's the first instance of "wx" in the file, to make finding it easy.

I am trouble-shooting some weirdness on Linux where the GUI for plugins do not load, and this may or may not be related to it .

### Steps to reproduce

It's a spelling error in the source file... Just open
/cmake-proxies/cmake-modules/AudacityDependencies.cmake
and search for  "wx"

### Expected behavior

Correct spelling?

### Actual behavior

wxWidgets spelled as wxWdidgets

### Audacity Version

current master / alpha / nightly build

### Operating system

Linux

**Comment by crsib** [comment]
I have grepped and only find wxWdidget inside the comment. Can you please give more details about the problems you are experiencing on Linux?

**Comment by evertvorster** [comment]
Hi there! 
I am still in the discovery phases of the issue, but can give you the outline already. 
As mentioned in the bug, I run Arch Linux. The officially packed audacity does not display the GUI of plugins. 

The packager is following Arch guidelines and is linking against the system wxWidgets. This is at version 3.2.0, and according to the build instructions for Audacity here on github the only version that works with Audacity is 3.1.3. Apparently a breaking change happened in wxWidgets 3.1.5 according to some comments I found from other people trying to build Audacity from source on Arch.

So, long term the version of wxWidgets that Audacity can use properly has to be updated, I suppose. If I can figure out what's holding back the working of Audacity against later versions of wxWidgets, there may be a future version of Audacity that works properly on Arch. 

For troubleshooting purposes I am building the debug version of Audacity from sources pulled from git master.
When I link against system wxWidgets, and try to open a Calf plugin, I don't get the nice graphics, and get a suil error in the console. The exact error I don't have now, that will be a problem for later when I investigate what is holding the version of wxWidgets back. Right now I just want something that works. 

When building against the supplied version of wxWidgets I do get a working GUI, although the wxWidgets that conan builds links to gtk2, and is as ugly as sin. Also, conan builds in my home directory, which is both surprising and totally unacceptable to me. 
I expect builds to only happen in the directory where the program sources reside. 

Also, as of 2022-08-27 vst3sdk does not build with conan, and if I install vst3sdk in the system the build scripts of Audacity does not detect it. 
vst3sdk from the system has a pkg-config file that does seem to work. Currently the only way to build Audacity from source is to completely disable vst3 plugins.

Moving forward, I'm not happy with the way conan works, and would like to ditch it completely. 

In Arch AUR there is an old version of wxWidgets (3.1.4) that is installed into /opt, and I want to see if I can link to that and get a working GUI for my plugins. Currently I am looking for a way to link to a specific version of wxWidgets installed in a custom location. Do you know of a way to instruct the build scripts to look for wxWidgets in a specific place?

Once I have a way of building Audacity on Arch that does not pull in external libraries and has a GUI for plugins I'll feed that back to the package maintainer on Arch. He can then decide whether it's worth it to install the old version of wxWidgets from AUR, or leave this version of Audacity without GUI's for plugins for this version.

Long term I would like to help this project a little by figuring out why wXWidgets is being held at a specific version, and why the build scripts are not detecting system .vst3sdk.

**Comment by evertvorster** [comment]
The suil error mentioned earlier:
suil error: Unable to wrap UI type <http://lv2plug.in/ns/extensions/ui#GtkUI> as type <http://lv2plug.in/ns/extensions/ui#Gtk3UI>

**Comment by evertvorster** [comment]
For future persons who would like to know the answer to how you make this build scripts link against wxWidgets in a custom location, here is the cmake flag that I used:
-DwxWidgets_CONFIG_EXECUTABLE=/opt/wxgtk-dev-314/bin/wx-config-gtk3

**Comment by crsib** [comment]
Thank you a lot for details

Master should build against 3.2.0 just fine, I am unaware of issues with LV2, I can check Tue. I will check for building against the system vst3sdk next week as well, hopefully. The current Conan supplied package does not build against GCC 12, we haven't had enough time yet to fix that.

In the meantime, please check https://github.com/audacity/audacity/blob/master/linux/packages/arch/PKGBUILD for an example of building against system libraries on arch. It also demostrates how to override Conan default locations (which are pretty common for the lnaguage-level package managing tools, I think). We do not override it by default to allow sharing the cache between different builds of Audacity and other software, that uses Conan.

**Comment by crsib** [comment]
Re: wxVersion upgrade: we have a huge amount of patches for wx 3.1 to support accessibility on macOS. Given that we are moving in the direction of changing the UI tookit it is not feasible right now to port those to wx 3.2. Linux, on the other hand, uses an unpatched version of wx 3.2.

**Comment by evertvorster** [comment]
Hi there!
Thanks so much for all the information. I'll check out the PKGBUILD now. 
For your information, when I build against 3.1.4 on the system, the error message is still the same. 
If the PKGBUILD provided also gives the same error message, then I'll open an issue for it. 
Thanks again for all the help!

**Comment by crsib** [comment]
I am quite confused with vst3sdk package provided by Arch Linux. It literally installs the source code for the SDK. What is the expected behavior for packages, depending on the vst3sdk?

**Comment by evertvorster** [comment]
Hi there! I am quite confused by is as well. I can reach out to the package
maintainer tonight.

Kind regards
